src/Draco.LanguageServer/Handlers/DracoSemanticTokensHandler.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using System.IO;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Draco.Compiler.Api.Syntax;
+using System.Reflection;
+
+namespace Draco.LanguageServer.Handlers;
+public class DracoSemanticTokensHandler : SemanticTokensHandlerBase
+{
+    private ParseTree? parseTree;
+
+    private readonly DocumentSelector documentSelector = new(new DocumentFilter
+    {
+        Pattern = $"**/*{Constants.DracoSourceExtension}",
+    });
+
+    public override async Task<SemanticTokens?> Handle(SemanticTokensParams request, CancellationToken cancellationToken)
+    {
+        var result = await base.Handle(request, cancellationToken).ConfigureAwait(false);
+        return result;
+    }
+
+    public override async Task<SemanticTokens?> Handle(SemanticTokensRangeParams request, CancellationToken cancellationToken)
+    {
+        var result = await base.Handle(request, cancellationToken).ConfigureAwait(false);
+        return result;
+    }
+
+    public override async Task<SemanticTokensFullOrDelta?> Handle(SemanticTokensDeltaParams request, CancellationToken cancellationToken)
+    {
+        var result = await base.Handle(request, cancellationToken).ConfigureAwait(false);
+        return result;
+    }
+
+    protected override async Task Tokenize(SemanticTokensBuilder builder, ITextDocumentIdentifierParams identifier, CancellationToken cancellationToken)
+    {
+        var content = await File.ReadAllTextAsync(DocumentUri.GetFileSystemPath(identifier)!, cancellationToken).ConfigureAwait(false);
+        this.parseTree = ParseTree.Parse(content);
+        var tokens = this.GetTokens(this.parseTree);
+        await Task.Yield();
+        foreach (var token in tokens)
+        {
+            builder.Push(Translator.ToLsp(token.Token.Range), token.Type, token.Modifiers);
```

Comment:
Wouldn't it make more sense to semantic token only contain a range, a type and modifiers if you are going to use it like this anyway? No need to store our token in it.